src/XrdHttp/XrdHttpError.hh

```diff
@@ -0,0 +1,79 @@
+#include <string>
+#include <sstream>
+#include "XProtocol/XProtocol.hh"
+#include "XrdHttpReq.hh"
+
+
+class XrdHttpError {
+    private:
+        // HTTP response parameters to be sent back to the user
+        int httpStatusCode;
+        std::string httpErrorCode;
+        std::string httpStatusText;
+        const char *desc;
+        const char *header_to_add;
+
+        static std::string formatStatusText(std::string errCode, std::string etext) {
+        return ("ERROR: " + errCode + ": " + etext);
+        }
+
+    
+    public:
+        XrdHttpError(int code, const std::string &errCode, const std::string &statusText, const char *desc, const char *header_to_add)
+        : httpStatusCode(code), httpErrorCode(errCode), httpStatusText(statusText), desc(desc), header_to_add(header_to_add) {}
+
+
+        static XrdHttpError Create( XResponseType xrdresp, XErrorCode xrderrcode, XrdHttpReq::ReqType httpVerb, XRequestTypes xrdOperation, std::string etext, const char *desc, const char *header_to_add){
+            int code{0};
+            std::string errCode;
+            std::string statusText;
+
+            switch (httpVerb){
+                case XrdHttpReq::rtPUT: 
+                    if (xrdOperation == kXR_open){
+                        if (xrderrcode == kXR_isDirectory) {
+                            code = 409;
+                            errCode = "8.1";
+                        }
+                        else if (xrderrcode == kXR_NoSpace) {
+                            code = 507;
+                            errCode = "8.3.1";
+                        }
+                        else if (xrderrcode == kXR_overQuota){
+                            code = 507;
+                            errCode = "8.3.2";
+                        }
+                        else if (xrderrcode == kXR_NotAuthorized){
+                            code = 403;
+                            errCode = "9.3";
+                        }
+                    }
+                    else if (xrdOperation == kXR_write){
+                        if (xrderrcode == kXR_NoSpace) {
+                            code = 507;
+                            errCode = "8.4.1";
+                        }
+                        else if (xrderrcode == kXR_overQuota){
+                            code = 507;
+                            errCode = "8.4.2";
+                        }
+                    }
+                    else {
+                        code = 500;
+                        errCode = "Unknown";
+                    }
+                    break;
+                default:
+                    code = 500;
+                    errCode = "Internal Server Error";
+                    break;
```

Comment:
This is confusing. Why have two distinct cases for 500? Can we let all 500 use `"Internal Server Error"`?